app/routes/app.$app.requests.tsx

```diff
@@ -51,12 +75,39 @@ export const meta: MetaFunction = () => {
 };
 
 export default function Requests() {
+	const [searchParams, setSearchParams] = useSearchParams();
+
+	const filters = React.useMemo<FilterForm>(
+		() => ({
+			query: searchParams.get("query") || "",
+			timeRange: (searchParams.get("period") as "45d" | "48h") || "45d",
+			maxDate: searchParams.get("start")
+				? new Date(searchParams.get("start")!)
+				: undefined,
+		}),
+		[searchParams],
+	);
+
+	const handleFilterChange = React.useCallback(
+		(updated: FilterForm) => {
+			const params: Record<string, string> = {
+				query: updated.query,
+				period: updated.timeRange,
+			};
+			if (updated.maxDate) {
+				params.start = updated.maxDate.toISOString().split("T")[0];
+			}
+			setSearchParams(params);
+		},
+		[setSearchParams],
+	);
+
 	return (
 		<>
 			<RequestsSum />
 			<RequestsGraph />
-			<Filter />
-			<RequestLogsTable />
+			<Filter onFilterChange={handleFilterChange} searchParams={searchParams} />
```

Comment:
Since you're getting `searchParams` from the hook, then just use the hook in `Filter` instead of passing it